File: exercises/day23-Portal-Stabilization/csharp/PortalStabilization/Models/KarmaBalance.cs
namespace PortalStabilization.Models;

public record KarmaBalance(
    string Dimension,
    int TotalKarmaDebt,
    int AnomalyCount)
{
    public double AverageKarmaDebt()
    {
        return AnomalyCount == 0 ? 0.0 : (double)TotalKarmaDebt / AnomalyCount;
    }
}

File: exercises/day23-Portal-Stabilization/java/src/main/java/craft/portal/KarmaBalance.java
package craft.portal;

public record KarmaBalance(
    String dimension,
    int totalKarmaDebt,
    int anomalyCount
) {
    public double averageKarmaDebt() {
        return anomalyCount == 0 ? 0.0 : (double) totalKarmaDebt / anomalyCount;
    }
}
